src/episodes.jl

```diff
@@ -0,0 +1,154 @@
+Base.length(es::EpisodesBuffer) = length(es.traces)
+Base.keys(es::EpisodesBuffer) = keys(es.traces)
+Base.keys(es::EpisodesBuffer{<:Any,<:Any,<:CircularPrioritizedTraces}) = keys(es.traces.traces)
+function Base.show(io::IO, m::MIME"text/plain", eb::EpisodesBuffer{names}) where {names}
+    s = nameof(typeof(eb))
+    t = eb.traces
+    println(io, "$s containing")
+    show(io, m::MIME"text/plain", t)
+end
+
+ispartial_insert(traces::Traces, xs) = length(xs) < length(traces.traces) #this is the number of traces it contains not the number of steps.
+ispartial_insert(es::EpisodesBuffer, xs) = ispartial_insert(es.traces, xs)
+ispartial_insert(traces::CircularPrioritizedTraces, xs) = ispartial_insert(traces.traces, xs)
+
+function fill_multiplex(es::EpisodesBuffer)
+    for trace in es.traces.traces
+        if !(trace isa MultiplexTraces)
+            push!(trace, last(trace)) #push a duplicate of last element as a dummy element, should never be sampled.
+        end
+    end
+end
+function fill_multiplex(es::EpisodesBuffer{<:Any,<:Any,<:CircularPrioritizedTraces})
+    for trace in es.traces.traces.traces
+        if !(trace isa MultiplexTraces)
+            push!(trace, last(trace)) #push a duplicate of last element as a dummy element, should never be sampled.
+        end
+    end
+end
+
+function Base.push!(eb::EpisodesBuffer, xs::NamedTuple)
+    push!(eb.traces, xs)
+    partial = ispartial_insert(eb, xs)
+    if length(eb.traces) == 0 && partial #first push should be partial
+        push!(eb.step_numbers, 1)
+        push!(eb.episodes_lengths, 0)
+        push!(eb.sampleable_inds, 0)
+    elseif !partial #typical inserting
+        if length(eb.traces) < length(eb) && length(eb) > 2 #case when PartialNamedTuple is used. Steps are indexable one step later
+            eb.sampleable_inds[end-1] = 1 
+        else #case when we don't, length of traces and eb will match.
+            eb.sampleable_inds[end] = 1 #previous step is now indexable
+        end
+        push!(eb.sampleable_inds, 0) #this one is no longer
+        ep_length = last(eb.step_numbers)
+        push!(eb.episodes_lengths, ep_length)
+        startidx = max(1,length(eb.step_numbers) - last(eb.step_numbers))
+        eb.episodes_lengths[startidx:end] .= ep_length
+        push!(eb.step_numbers, ep_length + 1)
+    elseif partial
+        fill_multiplex(eb)
+        eb.sampleable_inds[end] = 0 #previous step is not indexable because it contains the last state
+        push!(eb.sampleable_inds, 0) #this one isn't either
+        push!(eb.step_numbers, 1)
+        push!(eb.episodes_lengths, 0)
+    end
+    return nothing
+end
+
+function Base.push!(eb::EpisodesBuffer, xs::PartialNamedTuple) #wrap a NamedTuple to push without incrementing the step number. 
+    push!(eb.traces, xs.namedtuple)
+    eb.sampleable_inds[end-1] = 1 #completes the episode trajectory.
+end
+
+#= currently unsupported due to lack of support of appending a named tuple to traces with multiplextraces.
```

Comment:
Can we drop this code for now? Commented code always seems like a good idea, but is usually a pain to maintain (no tests, no CI, etc.)